### PR TITLE
BUG: Add warning when removing empty columns

### LINKED ISSUE #740: load_template_to_dataframe silently drops columns
If they are all `NA`, see [here](https://github.com/biocore/qiita/blob/master/qiita_db/metadata_template.py#L1748)

### PR DESCRIPTION
**Pull request by ElDeveloper**
A UserWarning is now raised whenever load_template_to_dataframe drops a column
if it is empty. This message is shown on the interface as well.

Fixes #740

### CODE REVIEW

File: qiita_db/test/test_metadata_template.py [hunk 1]
```diff
@@ -1481,6 +1482,21 @@ def test_load_template_to_dataframe_empty_sample_names(self):
         exp.index.name = 'sample_name'
         assert_frame_equal(obs, exp)
 
+    def test_load_template_to_dataframe_empty_column(self):
+
+            obs = npt.assert_warns(UserWarning, load_template_to_dataframe,
```
Comment (antgonza) [inline qiita_db/test/test_metadata_template.py]: Block is over indented, interesting that flake8 didn't complain.

File: qiita_pet/handlers/study_handlers/description_handlers.py [hunk 2]
```diff
@@ -280,9 +295,20 @@ def add_prep_template(self, study, user, callback):
...
+                # deleting previous uploads and inserting new one
+                pt_id = self.remove_add_prep_template(fp_rpt, raw_data_id,
+                                                      study, data_type_id,
+                                                      investigation_type)
+
+                # join all the warning messages into one. Note that this info
+                # will be ignored if an exception is raised
+                if warns:
+                    msg = '; '.join([str(w.message) for w in warns])
+                    msg_level = 'danger'
```
Comment (antgonza) [inline qiita_pet/handlers/study_handlers/description_handlers.py]: If this is a warning, should it be warning level vs. error?

Comment (antgonza) [inline qiita_pet/handlers/study_handlers/description_handlers.py]: While testing I realized that danger is actually the warning color so ignore.

**Comment by antgonza** [comment]
One comment that is not blocking so :+1:

**Comment by ElDeveloper** [comment]
I've addressed your comments, thanks! Note that the level was wrong,
should have been warning, not danger. Not sure why did it show you the
correct level. Maybe your browser cached the CSS value?

**Comment by ElDeveloper** [comment]
If someone else can review this PR, it should be ready for merge.

**Comment by josenavas** [comment]
@ElDeveloper which is the expected color of the message? From the code, it looks like it should be yellow (`warning`). However, when it shows up as `info` level (light blue):
screen shot 2015-02-05 at 5 44 27 pm

**Comment by josenavas** [comment]
Looks like the problem is [here] (https://github.com/biocore/qiita/blob/master/qiita_pet/templates/sitebase.html#L101) in sitebase.html

**Comment by ElDeveloper** [comment]
Thanks for catching that, I tested it but did not pay much attention to
the color and ended up blaming my browser. I'll fix this.

**Comment by josenavas** [comment]
Thanks @ElDeveloper It works perfect now:
screen shot 2015-02-05 at 6 20 57 pm

**Comment by ElDeveloper** [comment]
Thanks!

**Pull request merged**